packages/ember-glimmer/lib/syntax/curly-component.js

```diff
@@ -211,4 +231,18 @@ export class CurlyComponentDefinition extends ComponentDefinition {
     builder.attrs.dynamic('id', elementId);
     builder.attrs.static('class', 'ember-view');
   }
+
+  lateBound(template) {
+    let definition;
+    if (this._cache) {
+      definition = this._cache[template.id];
```

Comment:
given we create new instances from directly imported modules we likely should cache the instance we created for our env from the factory, otherwise the component will recreate and recompile everything if you directly import it onto layout.